purchase_sale_lot_intercompany/models/stock_production_lot.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# @author Guillaume MASSON <guillaume.masson@groupevoltaire.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockProductionLot(models.Model):
+    _inherit = "stock.production.lot"
+
+    def get_inter_company_lot(self, company):
+        self.ensure_one()
+        lot = self.sudo().search(
+            [
+                ("name", "=", self.name),
+                ("product_id", "=", self.product_id.id),
+                ("company_id", "=", company.id),
+            ]
+        )
+        if not lot:
+            lot = self.sudo().create(
```

Comment:
Not blocking comment as it is an improvment, but I think it would be nice to have a hook to get these vals.
Indeed, we can imagine cases where we add additional information on lot and we would like to propagate it.